bin/dfx-sns-disburse-neuron

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+SOURCE_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+PATH="$SOURCE_DIR:$PATH"
+. "$SOURCE_DIR/versions.bash"
+
+# Source the clap.bash file ---------------------------------------------------
+source "$SOURCE_DIR/clap.bash"
+# Define options
+clap.define short=n long=network desc="The dfx network to use" variable=DFX_NETWORK default="local"
+# Source the output file ----------------------------------------------------------
+source "$(clap.build)"
+set -euo pipefail
+
+GOVERNANCE_CANISTER_ID="$(dfx canister call sns_root list_sns_canisters '(record {})' | idl2json | jq -r '.governance | .[]')"
+
+NEURON_ID="$(dfx canister call "$GOVERNANCE_CANISTER_ID" list_neurons '(record {limit=10})' | idl2json | jq -r '.neurons[] | select(.dissolve_state[].DissolveDelaySeconds == "0") | .id[].id | join("; ")')"
```

Comment:
`list_neurons` returns all neurons. You probably want the ones belonging to the user only.

In this case you need to pass `of_principal` parameter.

See how we do it in ic-js [here](https://github.com/dfinity/ic-js/blob/cb3e651039e6ac7524237fa338a4c6711acce479/packages/sns/src/governance.canister.ts#L83).

Reply:
That's if you follow this process to participate in the swap.

But if you also participate manually, then you'll have more neurons, right?

Reply:
Ok, so the only way to have Get Tokens work would be if we don't manually participate in the sale?

That might be worth a comment somewhere then.

Reply:
Currently, the demo goes through the entire sale and finalizes it.
So if you run the demo, you can't participate in the sale.
If you run mksns manually, you can then participate and then you're on your own.
After you finalize the sale, you can manually transfer the tokens to the faucet if you want, or use the tokens from the neuron you got.
I haven't really thought about what would be useful or expected.
But the script isn't set in stone.
But it does now specify the of_principal as you suggested.

Reply:
Ok, thanks for the clarification.

I probably needed some context.